docs/sql-programming-guide.md

```diff
@@ -1058,6 +1058,20 @@ the Data Sources API. The following options are supported:
       The JDBC fetch size, which determines how many rows to fetch per round trip. This can help performance on JDBC drivers which default to low fetch size (eg. Oracle with 10 rows).
     </td>
   </tr>
+  
+  <tr>
+    <td><code>truncate</code></td>
+    <td>
+     This is a JDBC writer related option. To truncate the existing table before inserting the new data. This option only works with <code>SaveMode.Overwrite</code>. Without this option, Spark will drop the entire table, including its table definitions as well. <code>truncate</code> way is more efficient and ideal for cleaning out data from existing temp table. Its default value is <code>false</code>. 
```

Comment:
May I suggest somewhat different text here. I'd like to emphasize why you _wouldn't_ use it:

When `SaveMode.Overwrite` is enabled, this option causes Spark to truncate an existing table instead of dropping and recreating it. This can be more efficient, and prevents the table metadata (e.g. indices) from being removed. However, it will not work in some cases, such as when the new data has a different schema. It defaults to `false`.